convert-sg-db.py

```diff
@@ -0,0 +1,121 @@
+
+def standard_or_idea_to_html(document):
+    output = "\n <dt><a href=\"{0}\">{1}</a></dt>".format(document["href"], document["name"])
+    return output + "\n <dd>{}</dd>\n".format(commonmark.commonmark(document["description"])[:-1]) # Strip trailing \n
+
+def write_site_template(filename, template, title, content):
+    output = template.replace("@TITLE_GOES_HERE@", title)
+    output = output.replace("@CONTENT_GOES_HERE@", content)
+
+    open(filename, "w").write(output)
+
+
+def create_idea_whatwg_org(db, template):
+    content = "<p>The WHATWG works on a number of ideas that aspire to become web platform standards.</p>\n\n"
+    content += "<dl>"
+    for idea in db["ideas"]:
+        content += standard_or_idea_to_html(idea)
+    content += "</dl>\n\n"
+
+    write_site_template("idea.whatwg.org/index.html", template, "Ideas", content)
+
+
+def create_workstreams(db, template):
+    content = """<p>This document contains an official <a href="/sg-agreement#steering-group">Steering Group</a>-maintained accounting of the WHATWG workstreams, their responsible editors, and their associated standards.</p>
+
+<p>Note that per the <a href="/workstream-policy">Workstream Policy</a>, the official <a href="/workstream-policy#editor">Editors</a> listed here may have delegated responsibilities and editing duties to deputy editors for any given <a href="/workstream-policy#living-standard">Living Standard</a>. As such, the <a href="/workstream-policy#editor">Editor</a> position listed here is more about "who is the official liaison with the <a href="/sg-agreement#steering-group">Steering Group</a>" and less about "who is editing the document". To get an accounting of who is primarily responsible for day-to-day specification work, you'll be better served by checking the Acknowledgments section of the <a href="/workstream-policy#living-standard">Living Standard</a> in question.</p>
+
+<dl>"""
+    for workstream in db["workstreams"]:
+        content += "\n <dt>{}</dt>".format(workstream["name"])
+        content += "\n <dd>"
+        content += "\n  <dl>"
+        content += "\n   <dt>Scope</dt>"
+        content += "\n   <dd>{}</dd>".format(commonmark.commonmark(workstream["scope"])[3:-5]) # Strip leading <p> and trailing </p>\n
+        content += "\n   <dt>Editor{}</dt>".format("" if len(workstream["editors"]) == 1 else "s")
+        for editor in workstream["editors"]:
+            content += """\n   <dd>{0}, <a href="mailto:{1}">{1}</a></dd>""".format(editor["name"], editor["email"])
+        content += "\n   <dt>Standard{}</dt>".format("" if len(workstream["standards"]) == 1 else "s")
+        for standard in workstream["standards"]:
+            content += """\n   <dd><a href="{0}">{1}</a></dd>""".format(standard["href"], standard["name"])
+        content += "\n  </dl>"
+        content += "\n </dd>\n"
+    content += "</dl>\n\n"
+
+    write_site_template("whatwg.org/workstreams", template, "Workstreams", content)
```

Comment:
I think it would be nicer if all the I/O was in main() and the create_ functions were just pure functions that returned a string.